code/datums/world_topic.dm

```diff
@@ -96,6 +96,32 @@
 
 /datum/world_topic/news_report/Run(list/input)
 	minor_announce(input["message"], "Breaking Update From [input["message_sender"]]")
+	
+/datum/world_topic/ooc_relay
+	keyword = "ooc"
+	require_comms_key = TRUE
+
+/datum/world_topic/ooc_relay/Run(list/input)
+	var/messages = json_decode(input["message"])
+	var/oocmsg = messages["normal"]
+	var/oocmsg_toadmins = messages["admin"]
+	
+	var/source = json_decode(input["message_sender"])
+	var/sourceadmin = source["is_admin"]
+	var/sourcekey = source["key"]
+
+	//SENDING THE MESSAGES OUT
+	for(var/c in GLOB.clients)
+		var/client/C = c // God bless typeless for-loops
+		if( (C.prefs.chat_toggles & CHAT_OOC) && (sourceadmin || !(sourcekey in C.prefs.ignoring)) )
+			var/sentmsg // The message we're sending to this specific person
+			if(C.holder) // If they're an admin-ish
+				sentmsg = oocmsg_toadmins // Get the admin one
+			else
+				sentmsg = oocmsg
+			sentmsg = "<span class='prefix'>RELAY: [input["source"]]</span>" + sentmsg
```

Comment:
```suggestion
			sentmsg = "<span class='prefix'>RELAY: [input["source"]]</span> " + sentmsg
```

Reply:
why are you writting reviews on your own code...